### PR TITLE
Porting syntax-table-p

### LINKED ISSUE #1078: Port syntax-table-p
```c
DEFUN ("syntax-table-p", Fsyntax_table_p, Ssyntax_table_p, 1, 1, 0,
       doc: /* Return t if OBJECT is a syntax table.
Currently, any char-table counts as a syntax table.  */)
  (Lisp_Object object)
{
  if (CHAR_TABLE_P (object)
      && EQ (XCHAR_TABLE (object)->purpose, Qsyntax_table))
    return Qt;
  return Qnil;
}
```

### PR DESCRIPTION
**Pull request by mukuact**
Hi. I’m interested in this project and try to contribute!
This should resolve #1078

### CODE REVIEW

**Comment by shaleh** [comment]
Thanks @mukuact for pointing out a deficiency in our porting documentation. We used to have this more clearly spelled out but in moving things around I think we lost it.

**Comment by shaleh** [comment]
I updated this https://github.com/Wilfred/remacs/wiki/Porting-cookbook a few moments ago to document the `defsubr` changes needed.

File: rust_src/src/syntax.rs [hunk 1]
```diff
@@ -24,6 +24,16 @@ pub fn syntax_table() -> LispObject {
     ThreadState::current_buffer().syntax_table_
 }
 
+/// Return t if OBJECT is a syntax table.
+/// Currently, any char-table counts as a syntax table.
+#[lisp_fn]
+pub fn syntax_table_p(object: LispObject) -> bool {
+    object.is_char_table()
+        && object
```
[the remark this answers is not in the document]
Comment (brotzeit) [reply rust_src/src/syntax.rs]: Why not use `Option<LispCharTableRef>` as argument ?

[the remark this answers is not in the document]
Comment (brotzeit) [reply rust_src/src/syntax.rs]: I was wrong. It's ok.

**Comment by shaleh** [comment]
@mukuact, the porting cookbook talks about using Rust types instead of Lispobject where it makes sense. For this function, LispObject is the right option.

Look at the already ported DEFUN for good examples.

**Comment by brotzeit** [comment]
I shouldn't do anything while drinking my first coffee...

**Pull request merged**